### ISSUE TITLE
Update steps/description when teaching regex in Equation Solver Project

### ISSUE THREAD

**Issue by MarcinK00** [freeCodeCamp/freeCodeCamp#56353]
### Describe the Issue

I'm not sure of the author's intentions, but I think it would be helpful to direct the learner on how to represent multiple characters, as this was mentioned much earlier in the course. For example, in this manner:
"When referring to all digits, you can use regex special sequences or sets"

Related file - https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/07-scientific-computing-with-python/learn-interfaces-by-building-an-equation-solver/664ef4623946e65e18d59764.md

**Comment by icemberg** [comment]
The pattern (?<!\d) is a negative lookbehind assertion. It ensures that the character 1 is only substituted if it is not preceded by a digit.

**Comment by icemberg** [comment]
What if we return this one
 return re.sub(r'(?<!\d)1', '', equation_string.strip('+'));

**Comment by jdwilkin4** [comment]
Hi @MarcinK00 ! 

I have updated your issue to just focus on your second point you made about regex sequences. 
I have opened this up for the discussion so other contributors and the Python team can weigh in here.

As for the TODO comment, I have opened a separate issue for that since that can be handled pretty quickly and shouldn't be blocked by this ongoing discussion here 👍

**Comment by naomi-lgbt** [comment]
>  as this was mentioned much earlier in the course

That's kind of the idea. Our approach is often to get campers to recall (or revisit) concepts they have learned previously.

**Comment by MarcinK00** [comment]
I am closing the issue. I created it because the author of the code left a “TODO: add details/improve wording” comment that was displayed to the campers. The step doesn't seem too difficult based on the posts on FCC forum. So after removing the bug, I consider the problem solved.
